### PR TITLE
Initial commit for the VPC Builder

### PR DESCRIPTION
**Pull request by neoghostz**
VPC Builder example.

Includes test transform fragment JSON.

Extends on the VPC build with FlowLogs and IPv6 integration

### CODE REVIEW

**Comment by iann0036** [comment]
@neoghostz Could you add a link from the main README.md also?

File: VPCBuilder/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+It also adds in VPC Flowlogs with an IAM role and supports full dynamic allocation of IPv6 with the VPC and to each subnet. 
+
+The IPv6 handles Egress Internet Gateway and default route against ::/0
+
+## Build Package
+
+make buildPackage
+
+## Upload package to S3
+Fill in your bucket and profile (utilises a crud aws cli s3 upload command)
```
Comment (dbaggerman) [inline VPCBuilder/README.md]: I assume you meant "crude" here, not "crud".

File: VPCBuilder/README.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+## To Do
+
+Add Outputs with Exports for critical resources
+VPC Endpoints for all AWS Services
+Add a little better handling of custom pieces (e.g. different route gateways)
+Adding proper IPv6 regex and handling with NetworkACLs
+
+## Basic Usage
+
+Utilise the yaml structure below e a template, changing the Account ID in the transformation definiton.
```
Comment (dbaggerman) [inline VPCBuilder/README.md]: "below e a template" looks like another typo, presumably "below is a template".

**Comment by tlelson** [comment]
README.md looks fantastic @neoghostz .  Perhaps it is worth adding some discussion about the dependencies, namely the Virtual Private Gateway (VPG, in this case: vgw-0c003e11).  i.e. 

- what are the options for the user here to set this up?  
- What configurations of the VPG are supported/unsupported?  
- How best to build it.

File: VPCBuilder/README.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            NATGateways:
+                NATGW1:
+                    {Subnet: ReservedNet1, Routetable: InternalRT1}
+                NATGW2:
+                    {Subnet: ReservedNet2, Routetable: InternalRT2}
+                NATGW3:
+                    {Subnet: ReservedNet3, Routetable: InternalRT3}
+            NetworkACLs:
+                RestrictedSubnetAcl: 
+                    RestrictedSubnetAclEntryInTCPUnReserved: "90,6,allow,false,0.0.0.0/0,1024,65535"
```
Comment (ojkelly) [inline VPCBuilder/README.md:94]: Can we add some info to the readme, about what these positional args are?

**Comment by tlelson** [comment]
I couldn't get this work.  Maybe I missconfigured something.  

I did get it working but made some pretty drastic changes, sorry!
PR: https://github.com/KablamoOSS/cfn-macros/pull/2

File: VPCBuilder/README.md (hunk 2 above)
Comment (neoghostz) [reply VPCBuilder/README.md]: Corrected. Cheers.

File: VPCBuilder/README.md (hunk 3 above)
Comment (neoghostz) [reply VPCBuilder/README.md:94]: Added into the VPC Builder README.md

**Comment by neoghostz** [comment]
@eL0ck You'll need to change the VPGW to one you've got detached in your account. Its not under CFN control due to Direct Connect VIF implication if you want to tear up/down the VPC.

What error are you getting your ChangeSet when performing the Transform?

File: VPCBuilder/README.md (hunk 3 above)
Comment (ojkelly) [reply VPCBuilder/README.md:94]: 👍

**Pull request merged**